src/main/java/com/liefery/android/stop_badge/StopBadge.java

```diff
@@ -233,12 +233,15 @@ int shadowSizeX() {
     }
 
     int shadowSizeY() {
-        return (int) ( shadowRadius + Math.abs( shadowDx ) );
+        return (int) ( shadowRadius + Math.abs( shadowDy ) );
     }
 
     public Bitmap export( int size ) {
-        Bitmap bitmap = Bitmap.createBitmap( size + shadowSizeX() * 2, size
-            + shadowSizeY() * 2, Bitmap.Config.ARGB_8888 );
+        int newSize = Math.max( shadowSizeX(), shadowSizeY() ) * 2 + size;
```

Comment:
I think the `ImageView` not being square is not an issue.

If the `ScaleType` is not `CENTER` though, nothing will work anymore. But since we predefine this value, there should be no good reason to do so and we do not give any support for doing so. That is bat country.